### PR TITLE
Mistral Client Migration

### PR DESCRIPTION
**Pull request by bnativi**
Mistral released a 1.0.0 version of their client with some breaking changes. This PR migrates from mistral client version 0.*.* to >=1.0.0.

Only 4 of the file changes are directly related to the mistral client migration. The mistralai version was updated in the pre commit yaml and in the api pyproject.toml. The llm_clients.py and test_llm_clients.py files were updated to reflect the changes with the new mistralai version.

The other 5 file changes were made by Charles to fix some pre-commit issues that were indirectly introduced by the mistral client migration.

### CODE REVIEW

**Review by czaloom — DISMISSED** [review]

File: api/valor_api/schemas/metrics.py [hunk 1]
```diff
@@ -796,7 +796,7 @@ class BLEUMetric(BaseModel):
     """
 
     value: float
-    parameters: dict[str, str | list[int | float]]
+    parameters: dict[str, str | float | list[int | float]]
```
Comment (ntlind) [inline api/valor_api/schemas/metrics.py:799]: does float need to be listed twice here?

File: api/tests/functional-tests/backend/metrics/test_detection.py [hunk 2]
```diff
@@ -2292,11 +2292,11 @@ def test__convert_annotations_to_common_type(db: Session):
     mask[ymin:ymax, xmin:xmax] = True
 
     pts = [
-        (xmin, ymin),
-        (xmin, ymax),
-        (xmax, ymax),
-        (xmax, ymin),
-        (xmin, ymin),
+        (float(xmin), float(ymin)),
```
Comment (ntlind) [inline api/tests/functional-tests/backend/metrics/test_detection.py:2295]: is this change necessary? seems weird that updating the mistral client would require a change in this output

Comment (ntlind) [inline api/tests/functional-tests/backend/metrics/test_detection.py:2295]: why are these types changing if you're just updating the schema for LLM-related metrics?

Comment (czaloom) [reply api/tests/functional-tests/backend/metrics/test_detection.py:2295]: This is related to updating to the latest pre-commit version, for some reason the handling of typing `tuple[int | float, int | float]` has changed.

**Review by czaloom — APPROVED** [review]
